system/util/Util.cfc

```diff
@@ -111,4 +111,36 @@ component{
 		return slug;
 	}
 
-}	
\ No newline at end of file
+
+	/**
+	* Find all methods on a given metadata and it's parents with a given annotation
+	* @annotation The annotation name to look for on methods
+	* @metadata The metadata to search (recursively) for the provided annotation
+	*/
+    public array function getAnnotatedMethods(
+		required string annotation,
+		required struct metadata
+	) {
+        var lifecycleMethods = [];
+
+        if( StructKeyExists( arguments.metadata, "functions" ) ){
+            var funcs = arguments.metadata.functions;
+            lifecycleMethods.addAll( ArrayFilter( funcs, function( func ){
```

Comment:
Ok, I just discovered an issue with this.  `ArrayFilter` is only for cf10+ and UNFORTUNATELY, we will have to support cf9 for xunit support and the xunit runner is using it.  So we can either, rework this for cf9 compat: YUCK or we can only leave this feature for cf10+ only and remove it from the xunit runner.  Thoughts? @bdw429s @elpete